src/Mod/Fem/App/FemMesh.cpp

```diff
@@ -1940,121 +1945,222 @@ void FemMesh::writeABAQUS(const std::string& Filename, int elemParam, bool group
+            variants["Tria3"] = "M3D3";
+            variants["Quad4"] = "M3D4R";
+            variants["Tria6"] = "M3D6";
+            variants["Quad8"] = "M3D8R";
+            break;
+        case ABAQUS_FaceVariant::Stress:
+            variants["Tria3"] = "CPS3";
+            variants["Quad4"] = "CPS4";
+            variants["Tria6"] = "CPS6";
+            variants["Quad8"] = "CPS8";
+            break;
+        case ABAQUS_FaceVariant::Stress_Reduced:
+            variants["Tria3"] = "CPS3";
+            variants["Quad4"] = "CPS4R";
+            variants["Tria6"] = "CPS6";
+            variants["Quad8"] = "CPS8R";
+            break;
+        case ABAQUS_FaceVariant::Strain:
+            variants["Tria3"] = "CPE3";
+            variants["Quad4"] = "CPE4";
+            variants["Tria6"] = "CPE6";
+            variants["Quad8"] = "CPE8";
+            break;
+        case ABAQUS_FaceVariant::Strain_Reduced:
+            variants["Tria3"] = "CPE3";
+            variants["Quad4"] = "CPE4R";
+            variants["Tria6"] = "CPE6";
+            variants["Quad8"] = "CPE8R";
+            break;
+        case ABAQUS_FaceVariant::Axisymmetric:
+            variants["Tria3"] = "CAX3";
+            variants["Quad4"] = "CAX4";
+            variants["Tria6"] = "CAX6";
+            variants["Quad8"] = "CAX8";
+            break;
+        case ABAQUS_FaceVariant::Axisymmetric_Reduced:
+            variants["Tria3"] = "CAX3";
+            variants["Quad4"] = "CAX4R";
+            variants["Tria6"] = "CAX6";
+            variants["Quad8"] = "CAX8R";
+            break;
     }
 
+    // edge elements
+    switch (edgeVariant) {
+        case ABAQUS_EdgeVariant::Beam:
+            variants["Seg2"] = "B31";
+            variants["Seg3"] = "B32";
+            break;
+        case ABAQUS_EdgeVariant::Beam_Reduced:
+            variants["Seg2"] = "B31R";
+            variants["Seg3"] = "B32R";
+            break;
+        case ABAQUS_EdgeVariant::Truss:
+            variants["Seg2"] = "T3D2";
+            variants["Seg3"] = "T3D3";
+            break;
+        case ABAQUS_EdgeVariant::Network:
+            variants["Seg2"] = "B31";
+            variants["Seg3"] = "D";
+            break;
+    }
+
+    std::map<std::string, std::vector<int>> elemOrderMap;
+    std::map<int, std::string> edgeTypeMap;
+    std::map<int, std::string> faceTypeMap;
+    std::map<int, std::string> volTypeMap;
+
+
+    // node order fits with node order in importCcxFrdResults.py module to import
+    // CalculiX result meshes
+
+    // dimension 1
+    //
+    // seg2 FreeCAD --> B31, B31R, T3D2 CalculiX
+    // N1, N2
+    std::vector<int> seg2 = boost::assign::list_of(0)(1);
+    //
+    // seg3 FreeCAD --> B32, B32R, T3D3 D CalculiX
+    // N1, N3, N2
+    std::vector<int> seg3 = boost::assign::list_of(0)(2)(1);
+
+    elemOrderMap.insert(std::make_pair(variants["Seg2"], seg2));
+    edgeTypeMap.insert(std::make_pair(elemOrderMap[variants["Seg2"]].size(), variants["Seg2"]));
```

Comment:
Wouldn't these lines be equal? If they are, the proposed one looks simpler, IMO
```suggestion
    edgeTypeMap.insert(std::make_pair(seg2.size(), variants["Seg2"]));
```